### PR TITLE
Tidy up reusable actions and examples.

### PR DESCRIPTION
**Pull request by sengi**
- Document reusable actions in a consistent way.
- Mention in the top-level README that we have a library of reusable actions in this repo.
- Update to checkout v4 and turn off checkout progress logspew.
- Remove unnecessary `${{ }}` syntax from `if` conditions.
- Avoid unnecessarily overriding the name of the `checkout` step.
- Prune some comments that were just stating the obvious.

### CODE REVIEW

File: .github/workflows/brakeman.yml [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+#   push:
+#     branches: [main]
+#
+# jobs:
+#   security-analysis:
+#     name: Security Analysis
+#     uses: alphagov/govuk-infrastructure/.github/workflows/brakeman.yml@main
+
+# REUSABLE WORKFLOW
+# -----------------
```
Comment (theseanything) [inline .github/workflows/brakeman.yml:15]: I'm leaning toward removing these usage docs from all the workflows - I don't think they're helpful as you wouldn't use them verbatim, another thing to keep up to date, and it's not difficult to discern the usage from actual workflow definition.

File: README.md [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
 
-The govuk-infrastructure repo contains Terraform modules for turning up an EKS Kubernetes cluster for GOV.UK.
+The govuk-infrastructure repo contains:
 
-Also included are reusable workflows for running the code scanning tools Rubocop, Brakeman and the CodeQL Static Application Security Tests (SAST).
+- [`terraform/`](terraform/): Terraform modules for turning up an Kubernetes
+  cluster on EKS for GOV.UK.
+- [`.github/`](.github/): GitHub Actions and workflows used by other GOV.UK
+  repos, for example release automation, test runners and security analysis
+  tools.
```
Comment (theseanything) [inline README.md:11]: Dunno if we also wanna mention it has the Dockerfile/responsible for the build of the toolbox image: https://github.com/alphagov/govuk-infrastructure/blob/main/images/toolbox/Dockerfile

**Review by theseanything — APPROVED** [review]

File: .github/workflows/brakeman.yml (hunk 1 above)
Comment (sengi) [reply .github/workflows/brakeman.yml]: Done :)

File: README.md (hunk 2 above)
Comment (sengi) [reply README.md:11]: Ooh good catch - thanks!

Comment (sengi) [reply README.md:12]: Done :)

**Pull request merged**